spin/spin.go

```diff
@@ -65,15 +71,48 @@ func commandStart(command []string) tea.Cmd {
 		}
 
 		executing = exec.Command(command[0], args...) //nolint:gosec
-		if term.IsTerminal(os.Stdout.Fd()) {
+		executing.Stdin = os.Stdin
+
+		isTerminal := term.IsTerminal(os.Stdout.Fd())
+
+		//nolint:nestif
+		if isTerminal && runtime.GOOS == "windows" {
 			executing.Stdout = io.MultiWriter(&bothbuf, &outbuf)
 			executing.Stderr = io.MultiWriter(&bothbuf, &errbuf)
+			_ = executing.Run()
+		} else if isTerminal {
+			stdoutPty, err := openPty(os.Stdout)
+			if err != nil {
+				return errorMsg(err)
+			}
+			defer stdoutPty.Close() //nolint:errcheck
+
+			stderrPty, err := openPty(os.Stderr)
+			if err != nil {
+				return errorMsg(err)
+			}
+			defer stderrPty.Close() //nolint:errcheck
+
+			if outUnixPty, isOutUnixPty := stdoutPty.(*xpty.UnixPty); isOutUnixPty {
+				executing.Stdout = outUnixPty.Slave()
+			}
+			if errUnixPty, isErrUnixPty := stderrPty.(*xpty.UnixPty); isErrUnixPty {
+				executing.Stderr = errUnixPty.Slave()
+			}
+
+			go io.Copy(io.MultiWriter(&bothbuf, &outbuf), stdoutPty) //nolint:errcheck
+			go io.Copy(io.MultiWriter(&bothbuf, &errbuf), stderrPty) //nolint:errcheck
+
+			if err = stdoutPty.Start(executing); err != nil {
+				return errorMsg(err)
+			}
+			_ = xpty.WaitProcess(context.Background(), executing)
```

Comment:
Just an FYI, we don't need to use `xpty.WaitProcess` here, we can call `executing.Wait()` directly. That's what it does under the hood.

Reply:
@aymanbagabas I decided to keep as is so the code is more prepared to work on Windows if we decide to try again in the future. WDYT?